UserManagement/accounts/presentation/views.py

```diff
@@ -31,7 +30,44 @@ def update(self, request, *args, **kwargs):
     
 
 class InstructorRateViewSet(viewsets.ModelViewSet):
-    permission_classes = [HasViewInstructorRateClaim]
     queryset = InstructorRate.objects.all()
     serializer_class = InstructorRateSerializer
+    
+    def get_permissions(self):
+        if self.action == 'create':
+            permission_classes = [CanCreateInstructorRate]
+        elif self.action == 'destroy':
+            permission_classes = [CanDeleteInstructorRateAdminOnly]
+        else:
```

Comment:
What if "self.action" going to be some random stuff, do we still want to give them "HasViewInstructorRateClaim" ?

Reply:
Fixed , just interesting to ask Vitalik what he think

Reply:
I think we should be careful when assign the permissions, maybe better use switch case